### PR TITLE
Chore: clean up console warning from SQL Builder Editor tests

### PR DESCRIPTION
**Pull request by kevinwcyu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Relates #45024

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/components/SQLBuilderEditor/SQLBuilderEditor.test.tsx [hunk 1]
```diff
@@ -74,41 +74,6 @@ describe('Cloudwatch SQLBuilderEditor', () => {
...
-      expect(screen.getByLabelText('With schema')).toBeChecked();
-      expect(screen.getByText('Schema labels')).toBeInTheDocument();
-      await waitFor(() =>
-        expect(datasource.getDimensionKeys).toHaveBeenCalledWith(
-          query.namespace,
-          query.region,
-          { InstanceId: null },
-          undefined
-        )
-      );
```
[the remark this answers is not in the document]
Comment (yaelleC) [reply public/app/plugins/datasource/cloudwatch/components/SQLBuilderEditor/SQLBuilderEditor.test.tsx:108]: Removing the test does reduce unit test coverage a bit (59.8 to 58.3 for SQLBuilderEditor).

We could use the following to remove the warning:
```
it('Uses dimension filter when loading dimension keys', async () => {
    const query = makeSQLQuery({
      from: {
        type: QueryEditorExpressionType.Function,
        name: 'SCHEMA',
        parameters: [
          {
            type: QueryEditorExpressionType.FunctionParameter,
            name: 'AWS/EC2',
          },
          {
            type: QueryEditorExpressionType.FunctionParameter,
            name: 'InstanceId',
          },
        ],
      },
    });

    act(() => {
      render(<SQLBuilderEditor {...baseProps} query={query} />);
    });
    expect(screen.getByText('AWS/EC2')).toBeInTheDocument();
    expect(screen.getByLabelText('With schema')).toBeChecked();
    expect(screen.getByText('Schema labels')).toBeInTheDocument();
    await waitFor(() =>
      expect(datasource.getDimensionKeys).toHaveBeenCalledWith('AWS/EC2', query.region, { InstanceId: null }, undefined)
    );
  });
```

But then I see your point about testing an implementation detail which is not very RTL-like 🤔

[the remark this answers is not in the document]
Comment (andresmgot) [reply public/app/plugins/datasource/cloudwatch/components/SQLBuilderEditor/SQLBuilderEditor.test.tsx:108]: This is checking the API so it's still valuable, I wouldn't delete the test, it's better to test the interface than nothing.

Ideally, we could redefinge the mock for `getDimensionKeys` to verify the arguments were right and return something. As far as I can tell, these keys are later used in this `Select` (if I followed the code correctly)

https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/cloudwatch/components/SQLBuilderEditor/SQLBuilderSelectRow.tsx#L104

So we could verify that the final selector is rendered with the right options.

That's a bit out of the scope for the original task so going ahead with what @yaelleC suggests is also good enough.

[the remark this answers is not in the document]
Comment (kevinwcyu) [reply public/app/plugins/datasource/cloudwatch/components/SQLBuilderEditor/SQLBuilderEditor.test.tsx:108]: I've added in @yaelleC's suggested changes.

@yaelleC, how do I see the change in test coverage? I tried running `yarn test:coverage:changes` but I don't see any numbers similar to what you listed above. Is there another command?

**Review by andresmgot — APPROVED** [review]
LGTM

**Pull request merged**